### PR TITLE
feat(plugin-input-fields): add input fields for plugins

### PR DESCRIPTION
**Pull request by jatinriverlane**
## Changes
* Add input field for Experiment input
* Add input field for File input
* Add input field for TextArea input
* Add input field for Number input
* Add input field for Checkbox input
* Add input field for Dropdown input

### CODE REVIEW

File: aqueductcore/frontend/src/components/atoms/CheckboxField/index.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  padding: ${({ theme }) => theme.spacing(0.5)});
+  border-color: ${({ theme }) => theme.palette.mode === "dark" ? theme.palette.common.white : theme.palette.grey[600]};
+`;
+
+interface CheckboxProps {
+  title: string,
+  field: string,
+  description: string,
+  defaultValue?: string;
+}
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/atoms/CheckboxField/index.tsx:15]: I guess we can also use the same pattern here.
```suggestion
interface CheckboxProps extends PluginFieldBase{
  defaultValue?: string;
}
```

File: aqueductcore/frontend/src/components/atoms/FileField/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  padding: 0;
+  background-color: transparent;
+`;
+
+interface FileProps extends PluginFieldBase {
+  options: string[];
+  defaultValue?: string;
+}
+
+export function FileField({
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/atoms/FileField/index.tsx:20]: In this component by just passing this props
```JSX
    <FileField
        title="test"
        field="test"
        options={['1', '2']}
        description="test"
    />
```
I received this error
```error
Cannot read properties of null (reading 'focus')
```

File: aqueductcore/frontend/src/components/atoms/TextAreaField/index.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { Box, Grid, TextareaAutosize, styled } from "@mui/material";
+
+import { FieldDescription, FieldTitle, FieldType } from "components/atoms/sharedStyledComponents"
+import { PluginFieldBase } from "types/globalTypes";
+
+const TextAreaInput = styled(TextareaAutosize)`
+  resize: none;
+  width: 100%;
+  border-width: 1px;
+  border-color: ${({ theme }) => theme.palette.neutral.main};
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/atoms/TextAreaField/index.tsx:10]: This is not dark theme friendly, can you please also check others?

File: aqueductcore/frontend/src/components/atoms/NumberField/index.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  input[type=number] {
+    -moz-appearance: textfield;
+  }
+`;
+
+interface NumberProps extends PluginFieldBase {
+  defaultValue?: string;
+}
+
+export function NumberField({
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/atoms/NumberField/index.tsx:30]: Have you also handle INT and FLOAT difference here?

File: aqueductcore/frontend/src/components/atoms/sharedStyledComponents/index.tsx
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/atoms/sharedStyledComponents/index.tsx:1]: I still think it's a better idea to keep these in sepatated file, or if you think it's bette to have them in one place, maybe something like `sharedStyledComponents/InputFields/index.tsx`?

File: aqueductcore/frontend/src/components/atoms/CheckboxField/index.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+export function CheckboxField({
+  title = "",
+  field = "",
+  description = "",
+  defaultValue = ""
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/atoms/CheckboxField/index.tsx:21]: We can make consider them optional and keep empty string as default value, or if they're required no for the default value. By changing the type above I guess it mostly will be fixed, just we need to decide for `defaultValue` prop.

File: aqueductcore/frontend/src/components/atoms/ExperimentField/index.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+interface ExperimentProps extends PluginFieldBase {
+  experiment_title: string,
+  experiment_alias: string,
+}
+
+export function ExperimentField({
+  experiment_title,
+  experiment_alias,
+  description = "",
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/atoms/ExperimentField/index.tsx:30]: Same for default values or making them optional in type

**Review by safeamiiir — CHANGES_REQUESTED** [review]

File: aqueductcore/frontend/src/components/atoms/CheckboxField/index.tsx (hunk 1 above)
Comment (jatinriverlane) [reply aqueductcore/frontend/src/components/atoms/CheckboxField/index.tsx:15]: Yeah, missed this.

File: aqueductcore/frontend/src/components/atoms/CheckboxField/index.tsx (hunk 5 above)
Comment (jatinriverlane) [reply aqueductcore/frontend/src/components/atoms/CheckboxField/index.tsx:21]: Thanks @safeamiiir for help on this

**Review by safeamiiir — APPROVED** [review]

**Pull request merged**